internal/config/config_test.go

```diff
@@ -177,3 +181,57 @@ func TestConfig_LoadDirectory(t *testing.T) {
 	assert.Equal(t, pConfig, c.Inputs[3].Config,
 		"Merged Testdata did not produce correct procstat metadata.")
 }
+
+func TestConfig_LoadSpecialTypes(t *testing.T) {
+	c := NewConfig()
+	err := c.LoadConfig("./testdata/special_types.toml")
+	assert.NoError(t, err)
+	require.Equal(t, 1, len(c.Inputs))
+
+	inputHTTPListener, ok := c.Inputs[0].Input.(*http_listener_v2.HTTPListenerV2)
+	assert.Equal(t, true, ok)
+	assert.Equal(t, internal.Duration{Duration: time.Second}, inputHTTPListener.WriteTimeout)
+	assert.Equal(t, internal.Size{Size: 1024 * 1024}, inputHTTPListener.MaxBodySize)
+	assert.Equal(t, "/path/to/my/cert\n", inputHTTPListener.TLSCert)
+
+	c = NewConfig()
+	err = c.LoadConfig("./testdata/invalid_field.toml")
+	require.Error(t, err, "invalid field name")
+	// commented out asserts are differences in error output from influxdata/toml and naoina/toml
+	// assert.Equal(t, "Error parsing ./testdata/invalid_field.toml, line 2: field corresponding to `not_a_field' is not defined in `*http_listener_v2.HTTPListenerV2'", err.Error())
+	assert.Equal(t, "Error parsing ./testdata/invalid_field.toml, line 2: field corresponding to `not_a_field' is not defined in http_listener_v2.HTTPListenerV2", err.Error())
+
+	c = NewConfig()
+	err = c.LoadConfig("./testdata/wrong_field_type.toml")
+	require.Error(t, err, "invalid field type")
+	// assert.Equal(t, "Error parsing ./testdata/wrong_field_type.toml, line 4: http_listener_v2.HTTPListenerV2.Port: `string' type is not assignable to `int' type", err.Error())
+	assert.Equal(t, "Error parsing ./testdata/wrong_field_type.toml, line 4: (http_listener_v2.HTTPListenerV2.Port) cannot unmarshal TOML string into int", err.Error())
+
+	c = NewConfig()
+	err = c.LoadConfig("./testdata/wrong_field_type2.toml")
```

Comment:
Spend a little bit more time on naming the tests and input files.  A good test name will describe what is being tested, and a good test will check one behavior.